modules/decision/search_pattern.py

```diff
@@ -0,0 +1,242 @@
+        # search_gap_depth is the forwards/backwards distance
+        self.__search_width = (
+            2 * search_height * tan((camera_fov_sideways * pi / 180) / 2)
+        )
+        self.__search_depth = (
+            2 * search_height * tan((camera_fov_forwards * pi / 180) / 2)
+        )
+        self.__search_gap_width = self.__search_width * (1 - search_overlap)
+        self.__search_gap_depth = self.__search_depth * (1 - search_overlap)
+
+        self.__new_location = True
+
+        # Calculate positions for first square
+        self.__calculate_square_corners()
+        self.__calculate_side_of_square()
+
+    def __calculate_square_corners(self):
+        """
+        Computes the 4 corners of the current square
+        """
+        # Compute the size of square. First square will be a distance of the minimum of width and
+        # depth, each subsequent square will be a width larger
+        square_size = (
+            min(self.__search_gap_depth, self.__search_gap_width)
+            + (self.__current_square - 1) * self.__search_gap_width
+        )
+
+        # If the depth is less than the width, we apply an offset at each corner to ensure the
+        # entire area is scanned
+        adjustment = 0
+        if self.__search_gap_depth < self.__search_width:
+            adjustment = (self.__search_gap_width - self.__search_gap_depth) / 2
+
+        # Calculate the corners based on the offsets and the search origin. Top left corner is moved
+        # right by search_gap_width as the final side of the square will instead cover that part
+        self.__square_corners = [
+            (
+                self.__search_origin_x
+                - square_size
+                - adjustment
+                + self.__search_gap_width,
+                self.__search_origin_y + square_size,
+            ),  # Top left corner
+            (
+                self.__search_origin_x + square_size,
+                self.__search_origin_y + square_size + adjustment,
+            ),  # Top right corner
+            (
+                self.__search_origin_x + square_size + adjustment,
+                self.__search_origin_y - square_size,
+            ),  # Bottom right corner
+            (
+                self.__search_origin_x - square_size,
+                self.__search_origin_y - square_size - adjustment,
+            ),  # Bottom left corner
+        ]
+
+    def __calculate_side_of_square(self):
+        """
+        Computes the gaps along the current side of the square
+        """
+        # Calculate the positions of the current and next corners
+        self.__current_corner = self.__square_corners[self.__current_side_in_square]
+        next_corner = self.__square_corners[(self.__current_side_in_square + 1) % 4]
+
+        # Determine if the drone is moving horizontally or vertically along the current side
+        self.__moving_horizontally = self.__current_side_in_square % 2 == 0
```

Comment:
from above - having it hardcoded would make this more readable so we can tell which direction (clockwise vs counter clockwise) we're going